util/eventlog/loggables/loggables.go

```diff
@@ -26,3 +30,21 @@ func Error(e error) log.Loggable {
 		"error": e.Error(),
 	}
 }
+
+// Dial metadata is metadata for dial events
+func Dial(sys string, lid, rid peer.ID, laddr, raddr ma.Multiaddr) log.LoggableMap {
```

Comment:
to delay execution, you'll want to return a closure.

```Go
func Dial(....) eventlog.Loggable {
  return eventlog.LoggableF(func() map[string]interface{} {
  })
}
```

`LoggableF` is like `http.HandlerFunc`. It allows you to cast a function into a loggable.